M2/Macaulay2/d/ctype.d

```diff
@@ -52,9 +51,24 @@ export isspace    (c:char):bool := (chartype(c) & SPACE    ) != 0;
 export isnewline  (c:char):bool := (chartype(c) & NEWLINE  ) != 0;
 export isquote    (c:char):bool := (chartype(c) & QUOTE    ) != 0;
 
-export isvalidsymbol  (s:string):bool := (
-     if int(uchar(s.0)) == 226 && length(s) == 3 then return true; -- ugly unicode math symbol hack
+export ismathoperator(c1:int, c2:int):bool := (
+    c1 == 194      || -- latin-1 punctuation/symbols
+    c1 == 195 && (
+	c2 == 151  || -- \times
+	c2 == 183) || -- \div
+    c1 == 226);       -- mathematical operators
+
+-- number of bytes used to encode unicode character
+export numbytes(c:int):int := (
+    if      c < 192 then 1
+    else if c < 224 then 2
+    else if c < 240 then 3
+    else                 4);
+
+export isvalidsymbol(s:string):bool := (
      if !isalpha(s.0) then return false;
+     if (ismathoperator(int(s.0), int(s.1)) &&
+	 length(s) == numbytes(int(s.0))) then return true;
```

Comment:
does this mean say `getSymbol "⟎⟎"` is not a valid symbol? this used to cause an error but no longer now.